app/controllers/devices_controller.rb

```diff
@@ -0,0 +1,54 @@
+class DevicesController < ApplicationController
+
+  def index
+    @devices = Device.all
+  end
+
+  def new
+    @device = Device.new
+    @house = House.find(params[:house_id])
+    types = ["Fridge", "Oven", "Microwave", "Lamp", "Thermostat"]
+  end
+
+  def create
+    @house = House.find(params[:house_id])
+    @device = Device.new(user_params)
+    @device.house_id = @house.id
+    @device.save
+    redirect_to house_path(@house.id)
+  end
+
+  def show
+    @device = Device.find(params[:id])
+  end
+
+  def update
+    @device = Device.find(params[:id])
+    if @device.update(user_params)
+      redirect_to house_path
+    else
+      render 'edit'
+    end
+  end
+
+  def edit
+    @device = House.find(params[:id])
+  end
+
+  def destroy
+    @device = Device.find(params[:id])
```

Comment:
here for sure you should check if the device belongs to current_user houses. 
If you will live it in this way, I can delete others users devices. Same for House model